gifmodify.js

```diff
@@ -157,14 +158,76 @@ function addColorShiftingFramesGIF(inputGif, options, callback) {
 
   for (let i = 0; i < frames.length; i++) {
     let frame = frames[i];
-    setFrameProperties(frame);
+    setFrameProperties(frame, { interlaced: false });
     shiftColors(frame.bitmap, interval * i / frames.length, randomBlack, randomWhite);
   }
   callback(frames);
 }
 
-function setFrameProperties(frame) {
-  frame.interlaced = false;
+exports.createWigglingGIF = function(options) {
+  return new Promise((resolve, reject) => {
+    getGifFromBuffer(options.buffer).then(inputGif => {
+      addWigglingFramesGIF(inputGif, options, frames => {
+        let codec = new GifCodec();
+        codec.encodeGif(frames).then(resultGif => {
+          resolve(resultGif.buffer);
+        });
+      });
+    }).catch(error => reject(error));
+  });
+}
+
+function addWigglingFramesGIF(inputGif, options, callback) {
+  let imgWidth = inputGif.frames[0].bitmap.width;
+  let height = inputGif.frames[0].bitmap.height;
+  let width = imgWidth + 2 * Math.floor(imgWidth / 15); // ~6.6% of width is wiggle room for both sides
+  let margin = width - imgWidth;
+
+  let shiftSize = Math.max(1, Math.floor(margin / 6));
+  let interval = 2 * (margin / shiftSize + 4);
+  let stripeHeight = 2 * shiftSize;
+  let frames = alignGif(inputGif.frames, interval);
+
+  let startShift = margin / 2; // Initial offset of wiggle
+  let startLeft = true; // true -> go to left
```

Comment:
This is always true, make it an option?

Reply:
Starting off left or right only offsets the wiggle, still looks very similar so it wouldn't really matter
Start left: ![](https://cdn.discordapp.com/attachments/551812785575690261/642801920582352947/yentBlobXD.gif) Start right: ![](https://cdn.discordapp.com/attachments/551812785575690261/642802009401065482/yentBlobXD.gif)